examples/util.py

```diff
@@ -46,31 +46,35 @@ def run_ebtel(config, ebtel_dir):
         )
         if cmd.stderr:
             raise EbtelPlusPlusError(f"{cmd.stderr.decode('utf-8')}")
-        data = np.loadtxt(results_filename)
-
-        results = {
-            'time': data[:, 0]*u.s,
-            'electron_temperature': data[:, 1]*u.K,
-            'ion_temperature': data[:, 2]*u.K,
-            'density': data[:, 3]*u.cm**(-3),
-            'electron_pressure': data[:, 4]*u.dyne/(u.cm**2),
-            'ion_pressure': data[:, 5]*u.dyne/(u.cm**2),
-            'velocity': data[:, 6]*u.cm/u.s,
-            'heat': data[:, 7]*u.erg/(u.cm**3*u.s),
-        }
-
-        results_dem = {}
-        if config['calculate_dem']:
-            results_dem['dem_tr'] = np.loadtxt(
-                config['output_filename'] + '.dem_tr')
-            results_dem['dem_corona'] = np.loadtxt(
-                config['output_filename'] + '.dem_corona')
-            # The first row of both is the temperature bins
-            results_dem['dem_temperature'] = results_dem['dem_tr'][0, :]*u.K
-            results_dem['dem_tr'] = results_dem['dem_tr'][1:, :]*u.Unit('cm-5 K-1')
-            results_dem['dem_corona'] = results_dem['dem_corona'][1:, :]*u.Unit('cm-5 K-1')
-
-    return {**results, **results_dem}
+        
+        try:
+            data = np.loadtxt(results_filename)
+        except:
+            raise EbtelPlusPlusError("Failed to output a data file.\n")
```

Comment:
I'm a bit confused here as to why this isn't being printed to stderror. If so, I would assume it would be caught by the above if clause?

Reply:
I'm not sure.

The previous if statement is not catching an error for me.  The pytest fails on trying to read in the results file.  Here's what happens with the unedited code:
```
_________________________ test_minimum_heating[1e-15] _________________________

base_config = {'adaptive_solver_error': 1e-06, 'adaptive_solver_safety': 0.5, 'c1_cond0': 2.0, 'c1_rad0': 0.6, ...}
value = 1e-15

    @pytest.mark.parametrize('value', [-1e-5, 0, 1e-15])
    def test_minimum_heating(base_config, value):
        config = base_config.copy()
        config['background'] = value
        with pytest.raises(EbtelPlusPlusError):
>           run_ebtelplusplus(config)

tests/test_solver.py:84: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/helpers.py:20: in run_ebtelplusplus
    res = run_ebtel(config, REPO_DIR)
examples/util.py:51: in run_ebtel
    data = np.loadtxt(results_filename)
../../../anaconda3/lib/python3.11/site-packages/numpy/lib/npyio.py:1373: in loadtxt
    arr = _read(fname, dtype=dtype, comment=comment, delimiter=delimiter,
../../../anaconda3/lib/python3.11/site-packages/numpy/lib/npyio.py:992: in _read
    fh = np.lib._datasource.open(fname, 'rt', encoding=encoding)
../../../anaconda3/lib/python3.11/site-packages/numpy/lib/_datasource.py:193: in open
    return ds.open(path, mode, encoding=encoding, newline=newline)
```

Reply:
Ah, nevermind all that.  I'm just dumb.

The 'use_adaptive_solver' wasn't defined, so the code wasn't even running, hence no stderr, and it wasn't caught by the error handling.  

That raises one more point -- there should probably be a test that the config file has all the necessary variables defined.

Reply:
Agreed. The error handling in the C++ code is spotty at best. My ultimate plan is to do a better job at this when this all eventually gets wrapped in Python.